app/helpers/templates_helper.rb

```diff
@@ -28,6 +28,28 @@ def infobox_functions_and_macros(docs_section)
           :rel => 'external')
   end
 
+  def safemode_methods
+    @@safemode_methods ||= begin
+      objects = ObjectSpace.each_object(Class).select{|x| x < Safemode::Jail }
+      objects_with_methods = objects.map do |obj|
+        [obj.name.gsub(/::Jail$/, ''), obj.allowed_methods.sort.join(' ')]
+      end
+      objects_with_methods.uniq.sort_by(&:first)
+    end
+  end
+
+  def safemode_helpers
+    @@safemode_helpers ||= Foreman::Renderer::ALLOWED_HELPERS.sort.join(' ')
+  end
+
+  def safemode_variables
+    @@safemode_variables ||= Foreman::Renderer::ALLOWED_VARIABLES.sort.map{|x| "@#{x}"}.join(' ')
+  end
+
+  def infobox_safemode
+    alert(:class => 'alert-info', :header => '', :text => (_('In templates only selected methods and variables are allowed') + ', ' + link_to_function(_('list of available template helpers'), '$("#help_tab").click()')))
```

Comment:
I don't think this alert is really necessary. The "Help" tab is quite clear and visible, so this only uses up screen estate, I'd just remove it.